### PR TITLE
Remove need to send bank in retransmit request from ReplayStage

### PR DESCRIPTION
**Pull request by carllin**
#### Problem
No need to send bank in ReplayStage to the retransmitter because it's unused

#### Summary of Changes
Remove the bank from the Sender type

Fixes #

### CODE REVIEW

File: core/src/broadcast_stage.rs [hunk 1]
```diff
@@ -342,33 +343,33 @@ impl BroadcastStage {
...
 
         // Check for a retransmit signal
-        let mut retransmit_slots = retransmit_slots_receiver.recv_timeout(timer)?;
-        while let Ok(new_retransmit_slots) = retransmit_slots_receiver.try_recv() {
-            retransmit_slots.extend(new_retransmit_slots);
-        }
-
-        for (_, bank) in retransmit_slots.iter() {
-            let slot = bank.slot();
+        while let Ok(new_retransmit_slot) = retransmit_slots_receiver.try_recv() {
```
Comment (behzadnouri) [inline core/src/broadcast_stage.rs:351]: `Select::try_ready` might return with success spuriously.
I suggest do instead something like:
```rust
const RECV_TIMEOUT: Duration = Duration::from_millis(100);
let retransmit_slots =
    std::iter::once(retransmit_slots_receiver.recv_timeout(RECV_TIMEOUT)?)
        .chain(retransmit_slots_receiver.try_iter());

for new_retransmit_slot in retransmit_slots {
    ...
```

File: core/src/broadcast_stage.rs [hunk 2]
```diff
@@ -342,33 +343,33 @@ impl BroadcastStage {
...
     ) -> Result<()> {
-        let timer = Duration::from_millis(100);
+        let mut sel = Select::new();
+        sel.recv(retransmit_slots_receiver);
+        let _ = sel.ready_timeout(Duration::from_millis(100))?;
 
         // Check for a retransmit signal
-        let mut retransmit_slots = retransmit_slots_receiver.recv_timeout(timer)?;
-        while let Ok(new_retransmit_slots) = retransmit_slots_receiver.try_recv() {
-            retransmit_slots.extend(new_retransmit_slots);
```
Comment (jbiseda) [inline core/src/broadcast_stage.rs:350]: My reading of the original code was that it was trying to dedup multiple copies of the same bank being submitted in succession. Is that unnecessary?

File: core/src/broadcast_stage.rs (hunk 1 above)
Comment (carllin) [reply core/src/broadcast_stage.rs]: heh, and this is why you read the docs. Thanks for the suggestion!

File: core/src/broadcast_stage.rs (hunk 2 above)
Comment (carllin) [reply core/src/broadcast_stage.rs:350]: good catch, moved to a `HashSet` on the receiving end

**Review by jbiseda — APPROVED** [review]
thanks for making this more intuitive!

**Pull request merged**